website/blog/why-was-my-bazel-build-so-slow.md

```diff
@@ -0,0 +1,208 @@
+- Did Bazel discard its analysis cache due to a configuration change?
+  - To avoid corruption, Bazel will discard the analysis cache if a build has a different build option or configuration from previous builds. When this happens, you’ll see a log line like `WARNING: Build option --compilation_mode has changed, discarding analysis cache`.
+  - For example if you use the flag `--compilation_mode fastbuild` on a series of builds and `--compilation_mode opt` on a later build, you will notice that the later build is slower because the server will have discarded the analysis cache and must restart the analysis phase from scratch.
+
+If you’re using Bazel in CI, you may be frustrated by the lack of good options to save the analysis cache between runs on remote CI runners. Our CI solution, [BuildBuddy Workflows](https://www.buildbuddy.io/blog/meet-buildbuddy-workflows/), preserves the analysis cache by cloning CI runners to guarantee the in-memory state is not lost. Check out our [BazelCon](https://www.youtube.com/watch?v=YycEXBlv7ZA&list=PLxNYxgaZ8Rsefrwb_ySGRi_bvQejpO_Tj&index=10) talk for more technical details.
+
+⭐ If your build was slow due to fetching external dependencies, you may want to investigate the [repository cache](https://sluongng.hashnode.dev/bazel-caching-explained-pt-3-repository-cache). During a clean build, Bazel will fetch the external dependencies referenced in your WORKSPACE file and store the downloaded blobs in the repository cache. If the dependencies have not changed between builds, Bazel will not need to re-download them.
+
+The repository cache is configured by default and its location can be set with `--repository_cache`. It is not wiped with `bazel clean`, but will get wiped with `bazel clean --expunge`.
+
+⭐ You may also want to consider how many threads are working on analysis / dependency fetching. Allocating more CPUs to the build and parallelizing it further may result in performance gains.
+
+In the example images included above, you can see that only two threads are working on analysis. Bazel tries to determine an appropriate number of threads based on the available CPU cores on your machine, but you can manually set it with the `--jobs` flag.
+
+## 3) Does Bazel have enough memory?
+
+Famously, Bazel loves memory. This section is short and sweet, but describes a common problem we’ve seen with our customers.
+
+⭐ If Bazel doesn’t have enough memory, it will evict parts of the analysis cache. As described in the previous section, this can significantly slow down builds. If you notice a lot of `MerkleTree.build` actions even when you’re reusing a warm Bazel cache or see heavy garbage collection activity in the timing profile, you may want to increase the amount of RAM available to Bazel by increasing the JVM heap size with `--host_jvm_args`.
```

Comment:
Do we have a value we usually recommend, even if it's relative to the system properties? Or if not, maybe it's worth mentioning what the default is so it's obvious where to start?